pkg/cluster/loadbalancerprofile.go

```diff
@@ -174,48 +193,79 @@ func addOutboundIPsToLB(resourceGroupID string, lb mgmtnetwork.LoadBalancer, obI
 // The default outbound ip is saved if the api server is public.
 func (m *manager) deleteUnusedManagedIPs(ctx context.Context) error {
 	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	ipsToDelete, err := m.getManagedIPsToDelete(ctx)
```

Comment:
I like that name change, I'll update this.

My reason for the additional function was to try and keep the functions at a reasonable length.  If I combined these functions it would result in about 60 lines of code.  Instead I opted to keep the logic separate and used a function name to convey the logic to the reader.

Reply:
My humble and personal opinion is that the separate function helps a lot to understand the code. I am a strong advocate of small functions. 🙂